etl/steps/data/garden/ember/2022-12-13/combined_electricity.py

```diff
@@ -0,0 +1,361 @@
+        _combined = pd.concat([_global_data, _european_data], ignore_index=True)
+        # On rows where both datasets overlap, give priority to european review data.
+        _combined = _combined.drop_duplicates(subset=index_columns, keep="last")
+        # Combine data for different variables.
+        combined = pd.merge(combined, _combined, on=index_columns, how="outer")
+    error = "There are repeated columns when combining global and european review tables."
+    assert len([column for column in combined.columns if column.endswith("_x")]) == 0, error
+
+    # Create a table (with no metadata) and sort data appropriately.
+    combined = catalog.Table(combined).set_index(index_columns).sort_index()
+
+    # Ensure all column names are snake lower case.
+    combined = catalog.utils.underscore_table(combined)
+
+    # Import metadata from metadata yaml file.
+    combined.update_metadata_from_yaml(METADATA_PATH, "combined_electricity")
+
+    return combined
+
+
+def create_net_flows_table(ds_european: catalog.Dataset) -> catalog.Table:
+    """Create a table for the net flows of the European Electricity Review, since it has a different structure and
+    cannot be combined with the rest of tables in the Global or European Electricity Reviews.
+
+    Parameters
+    ----------
+    ds_european : catalog.Dataset
+        European Electricity Review.
+
+    Returns
+    -------
+    net_flows : catalog.Table
+        Table of net flows.
+
+    """
+    # Keep the net flows table as in the original european review (this table was not present in the global review).
+    net_flows = ds_european["net_flows"].copy()
+    # Import metadata from metadata yaml file.
+    net_flows.update_metadata_from_yaml(METADATA_PATH, "net_flows")
+
+    return net_flows
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load data.
+    #
+    # Read global and european electricity datasets from garden.
+    ds_global = catalog.Dataset(GLOBAL_DATASET_PATH)
+    ds_european = catalog.Dataset(EUROPEAN_DATASET_PATH)
+
+    #
+    # Process data.
+    #
+    # Combine all tables of the yearly electricity data into one.
+    combined_global = combine_yearly_electricity_data(ds_global=ds_global)
+
+    # Combine all tables of the european electricity review into one.
+    combined_european = combine_european_electricity_review_data(ds_european=ds_european)
+
+    # Combine global and european reviews.
+    combined = combine_global_and_european_electricity_review_data(
+        combined_global=combined_global, combined_european=combined_european
+    )
+
+    # Create an additional table with the electricity net flows (only available in european review).
+    net_flows = create_net_flows_table(ds_european=ds_european)
+
+    #
+    # Save outputs.
+    #
+    ds_garden = catalog.Dataset.create_empty(dest_dir)
+    # Import metadata from the metadata yaml file.
+    ds_garden.metadata.update_from_yaml(METADATA_PATH, if_source_exists="replace")
```

Comment:
You could use the new method: `ds_garden.update_metadata(METADATA_PATH)`. This also affects the building of all tables in `combine_global_and_european_electricity_review_data`, `combine_yearly_electricity_data`, `combine_european_electricity_review_data` and `create_net_flows_table`.

Note that you should call this function after adding the tables to the dataset.